core/templates/services/math-interactions.service.spec.ts

```diff
@@ -205,6 +214,77 @@ describe('MathInteractionsService', () => {
     expect(mathInteractionsService.getWarningText()).toBe(
       'Your answer contains an invalid term: 3.4.5');
 
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a4', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer contains invalid variable multiplication: a4. ' +
+      'The variable should come after the number: 4a.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a * 4', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer contains invalid variable multiplication: a*4. ' +
```

Comment:
I don't think this is wrong, actually. The problem only arises when they're concatenated.

Reply:
Done.